rules/apps.libsonnet

```diff
@@ -63,20 +63,18 @@
           {
             record: 'namespace:kube_pod_container_resource_requests_memory_bytes:sum',
             expr: |||
-              sum by (namespace, label_name) (
+              sum by (namespace) (
                   sum(kube_pod_container_resource_requests_memory_bytes{%(kubeStateMetricsSelector)s} * on (endpoint, instance, job, namespace, pod, service) group_left(phase) (kube_pod_status_phase{phase=~"Pending|Running"} == 1)) by (namespace, pod)
-                * on (namespace, pod)
-                  group_left(label_name) kube_pod_labels{%(kubeStateMetricsSelector)s}
+                * max by(namespace, pod) (kube_pod_labels{%(kubeStateMetricsSelector)s})
```

Comment:
We don’t need this at all now, as the value of this metric is always 1. We do need to make sure the sum above doesn’t still have the same problem.

Reply:
AFAICT we still need this to avoid many-to-many errors if `kube_pod_labels` has multiple timeseries for the same (namespace, pod) combination (as reported in https://bugzilla.redhat.com/show_bug.cgi?id=1779324).

Reply:
What I'm saying is that this multiplication is a no-op now. The value of the `kube_pod_labels` metric is always 1, and we're not joining any labels anymore, so this is not needed at all anymore.

Reply:
:man_facepalming: